### PR TITLE
Add views option

### PR DESCRIPTION
**Pull request by isohi**
I want to customize view.

### CODE REVIEW

File: lib/carraway/config.rb [hunk 1]
```diff
@@ -6,6 +6,8 @@ class << self
       def load(file)
         @config = YAML.load_file(file)
         Category.load(@config['categories'])
+
+        @root = ::File.expand_path('..', file)
```
Comment (adorechic) [inline lib/carraway/config.rb]: Custom view option determines configuration file path as a view root path?
I think this behavior is confusing and easy to cause unexpected behavior.

File: lib/carraway/server.rb [hunk 2]
```diff
@@ -6,11 +6,17 @@
...
-    set :views, ::File.expand_path('../views', __FILE__)
     set :method_override, true
     enable :sessions
     use Rack::Flash
 
+    def view(template)
+      if Config.views && ::File.exists?("#{Config.views}/#{template}.erb")
+        return erb template, views: Config.views
+      end
+      erb template
```
Comment (adorechic) [inline lib/carraway/server.rb]: Does this code path take a correct template path without `views` option?

File: carraway.gemspec [hunk 3]
```diff
@@ -27,7 +27,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'rack-flash3'
   spec.add_dependency 'redcarpet'
 
-  spec.add_development_dependency "bundler", "~> 1.16"
+  spec.add_development_dependency "bundler", "~> 2.0"
```
Comment (adorechic) [inline carraway.gemspec]: Does this enhancement require bundler 2? If you want to just update, I think you should make another pull request.

File: spec/test_views.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+categories:
```
Comment (adorechic) [inline spec/test_views.yml]: I think this test configuration and custom view should be put on example or fixture directory.
Putting similar test file on same directory is not developer friendly.

File: carraway.gemspec (hunk 3 above)
Comment (isohi) [reply carraway.gemspec]: This enhancement doesn't require bundler 2. I'll make another pull request.

File: lib/carraway/server.rb (hunk 2 above)
Comment (isohi) [reply lib/carraway/server.rb]: Yes.
There is no problem because sinatra is set by default.
https://github.com/sinatra/sinatra/blob/2e368238238358538cc0e02fe900f3ea05705c0a/lib/sinatra/base.rb#L1828-L1829

File: lib/carraway/config.rb (hunk 1 above)
Comment (isohi) [reply lib/carraway/config.rb]: I think so, so I use carraway working directory as a view root path.

File: spec/test_views.yml (hunk 4 above)
Comment (isohi) [reply spec/test_views.yml]: I think so, so I moved views option test to fixture directory.

**Comment by isohi** [comment]
[PTAL] @adorechic

File: lib/carraway/config.rb [hunk 5]
```diff
@@ -27,6 +27,10 @@ def file_backend
       def labels
         @config['labels'] || {}
       end
+
+      def views
+        @config['views'] && "#{Carraway.working_dir}/#{@config['views']}"
```
Comment (adorechic) [inline lib/carraway/config.rb]: This implementation does not support fullpath.
I prefer
```ruby
@config['views'] && File.expand_path(Dir.new(@config['views']))
```

File: README.md [hunk 6]
```diff
@@ -79,6 +79,14 @@ labels:
   label_key: 'Label title'
 ```
 
+### Configuring views
+If you would like to configure views, use `views` option.
+Add `views` to carraway.yml and copy `lib/carraway/views` directory to your project.
+
+```yaml
+views: 'views'
```
Comment (adorechic) [inline README.md]: How about `views_dir` ?

And `views_dir: 'path/to/your/views'`

File: README.md [hunk 7]
```diff
@@ -79,6 +79,14 @@ labels:
   label_key: 'Label title'
 ```
 
+### Configuring views
```
Comment (adorechic) [inline README.md]:
```
### Customized views
You can use your erb file as carraway views. 
If `views_dir` option is given, Carraway switches view root directory and use your erb file.
Carraway just ignores unknown or different erb files.
Please see default erb in https://github.com/adorechic/carraway/tree/master/lib/carraway/views

**Comment by isohi** [comment]
[PTAL] @adorechic
